source/components/NumericInput.js

```diff
@@ -151,22 +161,53 @@ class NumericInput extends Component<Props, State> {
     }
   }
 
+  _validateLimitProps(minValue: number, maxBeforeDot: number, maxAfterDot: number) {
+    const maxBeforeDotIsNum = maxBeforeDot && typeof maxBeforeDot === 'number';
+    const maxAfterDotIsNum = maxAfterDot && typeof maxAfterDot === 'number';
+    // if minValue is a float, it will split at the decimal
+    // trailing zeros are dropped with parseFloat
+    const minValParts = parseFloat(minValue).toString().split('.');
+
+    // if minValParts array has length of 2, it is a float
+    if (minValParts.length >= 2) {
+      const minValBeforeDot = minValParts[0];
+      const minValAfterDot = minValParts[1];
+
+      // if the number of integers in minValue is greater than maxBeforeDot
+      if (maxBeforeDotIsNum && (minValBeforeDot.length > maxBeforeDot)) {
+        // the combo is incompatible, throw error
+        const error = `
+          minValue: ${minValue} exceeds the limit of maxBeforeDot: ${maxBeforeDot}.
+          Adjust the values of these properties.
+        `;
+        throw new Error(error);
+      // if the number of decimal spaces in minValue is greater than maxBeforeDot
+      } else if (maxAfterDotIsNum && (minValAfterDot.length > maxAfterDot)) {
+        const error = `
+          minValue: ${minValue} exceeds the limit of maxAfterDot: ${maxAfterDot}.
+          Adjust the values of these properties.
+        `;
+        throw new Error(error);
```

Comment:
@nikolaglumac Here is how this looks from the user's perspective:
<img width="407" alt="screen shot 2018-05-09 at 11 26 33 pm" src="https://user-images.githubusercontent.com/13347886/39856899-6a70155a-53e6-11e8-9239-eee7b31983cb.png">
<img width="548" alt="screen shot 2018-05-09 at 11 24 34 pm" src="https://user-images.githubusercontent.com/13347886/39856924-7ef30960-53e6-11e8-81b3-9fa2c84c17a5.png">